crates/iceberg/src/expr/bound.rs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! This module defines bound expressions.
+
+use crate::expr::Operator;
+use crate::spec::{Literal, NestedFieldRef};
+
+/// Bound expression.
+pub enum Bound {
```

Comment:
The only thing I'm unsure about is having a Literal as a bound. It feels counterintuitive to me since you cannot bind a literal.

Reply:
This is for completeness of the model of an expression.

Reply:
Should we follow a model closer to what's done in the Java library? Even though literals are technically the leaf they don't extend Expression.

Reply:
Seems like this model is not familiar to people in iceberg community. I think it's worth to refactor it to a model easier to read.